src/plugin.js

```diff
@@ -33,15 +33,27 @@ module.exports = {
         redo() {
           let commit = this.undone.pop();
           this.newMutation = false;
-          this.$store.commit(`${commit.type}`, Object.assign({}, commit.payload));
+          switch (typeof commit.payload) {
```

Comment:
This would be better if it was a function, so you'll avoid duplication of logic

Reply:
Agreed. I'll merge the PR, though, and update it.